fingerprint/src/lib.rs

```diff
@@ -316,6 +317,79 @@ pub fn fingerprint_stream<R: BufRead + Send + Seek + 'static>(
     })
 }
 
+/// The result of eagerly running all fingerprint [`Kind`]s on some given content.
+///
+/// This structure is equivalent to [`Combined`], but each fingerprint is a tuple of the computed fingerprint
+/// plus the content that was processed to make the fingerprint.
+#[derive(Clone, Hash, Eq, PartialEq, Debug, Getters, Serialize, Deserialize)]
+#[getset(get = "pub")]
+pub struct Processed {
```

Comment:
At a glance, maybe an enum/sum type might be more apt?

```rust
enum Processed {
  Binary { raw: ... }
  NonBinary  { raw: ..., without_comments: .... }
}
```

Reply:
🤔 Maybe... the issue with that idea is that the presence of a `comment_stripped` `Some` value is somewhat orthogonal to whether the file was detected as binary (as is demonstrated by the `fingerprints_binary_file_appearing_as_text` [test here](https://github.com/fossas/foundation-libs/blob/5084976bd281f81e304075c0e1dee280080dc009/fingerprint/src/tests.rs#L96-L104)).

So really doing this would just work around the use of `Option`, which I don't know is actually very useful.